### ISSUE TITLE
Crash on windows 10 running Intel processor/graphics card

### ISSUE THREAD

**Issue by marza91** [qutebrowser/qutebrowser#5067]
**Version info (see `:version`)**:
https://paste.the-compiler.org/view/b2fc0322

**Does the bug happen if you start with `--temp-basedir`?**:
Yes

**Description**
Running qutebrowser installed from chocolatey on windows 10, I sometimes get this issue, randomly when opening a new page or tab:

```
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile) .
This is most likely caused by not having the necessary graphics drivers installed.

Install a driver providing OpenGL 2.0 or higher, or, if this is not possible, make sure the ANGLE Open GL ES 2.0 emulation libraries (libEGL.dll, libGLESv2.dll and d3dcompiler_*.dll) are available in the application executable's directory or in a location listed in PATH.
```

Some of the times the page I'm currently on will freeze up beforehand, but the crash only happens when I try to navigate in the current window or open a new tab. I know this because I can still run commands and keyboard shortcuts, but the GUI just stops responding completely. I've read some previous issues on similar issues on linux using an Intel Graphics card, so I'm suspecting this could be related? It's an Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz with a Intel(R) HD Graphics 620 display adapter. It's running the latest available driver and is supposed to support OpenGL 4.6.0, so I don't think the driver is the issue here(?).

Oh, and I've been running Arch on this machine previously with qutebrowser installed there as well with no issues at all. I had to install Windows recently for some work related stuff, and it's been like this from day one. I also tried to do remove config, data and cache folders, but the issue persisted.

Would love to figure this out, so some further troubleshooting instructions would be highly appreciated. I might be able to see if I can build/debug the situation later.

**How to reproduce**
Start qutebrowser normally (or with --temp-basedir) and start browsing. After a while, sometimes after 2-3 navigations, sometimes after 5-10 minutes the issue occurs randomly.

**Comment by The-Compiler** [comment]
Could you try using [Falkon](https://www.falkon.org/) for a while and see if you run into the same issue?

**Comment by marza91** [comment]
Just did a test, and got the exact same message. So I'm guessing this is related to QtWebEngine then?

I also tried setting webkit as backend, but it seems I can't start the browser that way. Do I need to install a dependency for that, or is webkit not supported for windows?

**Comment by The-Compiler** [comment]
Yep, it's a bug with either QtWebEngine or your graphic drivers then. Some ideas of what you could do:

- See if you see the same issue with Chromium as well
- Test the `qt.force_software_rendering` option (set to either `chromium`, `qt-quick` or `software-opengl`) - note that it needs a restart
- Wait until I release qutebrowser v1.9.0 with Qt 5.13 and see how things look there (might be in around two weeks already)
- If you don't see it with Chromium, report it to the [Qt bugtracker](https://bugreports.qt.io/secure/Dashboard.jspa)

To use WebKit on Windows, you'd need to build Qt/WebKit/PyQt by hand, which unfortunately is quite a challenge.

**Comment by marza91** [comment]
Thanks for the info!

Setting force_software_rendering to chromium seems to have fixed the issue for now.
Also running chromium 75 seem to work without issue, so I'll try to remove the option again once 1.9.0 comes out.

**Comment by marza91** [comment]
Never mind, the rendering option didn't help after all, just took a while for it to happen again :( 
Will keep trying the other options and look into raising an issue with QT if it doesn't resolve itself after an update.

**Comment by johnfarina** [comment]
@marza91 I'm in the same boat. Setting `c.qt.force_software_rendering = 'qt-quick'` seems to avoid this bug for me (on a Surface Pro 6), at least for the past 12 hours or so, whereas with this set to either `chrome` or `software-opengl` (or the default) qutebrowser would crash every few minutes. Maybe give that a try?

**Comment by marza91** [comment]
@johnfarina That's what I ended up doing as well, and it seems to be running great. Didn't want to post before I was more sure than last time, but I've been running it since yesterday without the crash.

Only issue I've found is that video playback seems to not work, but that's not really an issue for me. I've also started looking into going the WSL+VcXsrv route, and I just managed to get qutebrowser started there as well! 😄

**Comment by johnfarina** [comment]
@marza91 You are correct, video mostly isn't working for me either. Not to derail this issue, but how did you get qutebrowser running under WSL? I've got VcXsrv running and configured to handle X window applications correctly, but launching qutebrowser gives me a bunch of errors:
```
shared memfd open() failed: Function not implemented
19:24:43 INFO: Run :adblock-update to get adblock lists.
19:24:43 WARNING: QQuickWidget: Failed to make context current
19:24:43 WARNING: QQuickWidget::resizeEvent() no OpenGL context                                                        19:24:43 WARNING: QQuickWidget: Failed to make context current                                                         19:24:43 WARNING: QQuickWidget: Attempted to render scene with no context
19:24:44 WARNING: composeAndFlush: makeCurrent() failed
19:24:44 WARNING: composeAndFlush: makeCurrent() failed   
...
```

**Comment by The-Compiler** [comment]
@johnfarina When I tried a couple of days ago, an Ubuntu-installed Qt worked, but a newer Qt installed from PyPI (with a tox qutebrowser install) showed something similar like that.

**Comment by marza91** [comment]
@johnfarina I'm able to run the apt package version, which is 1.1.1 on Ubuntu 18.04 LTS. I managed to get the latest version from git to build and run as well, but I had to install qt from apt first. Haven't seen the issue you experienced at all.
That being said, everything runs a bit slowly for me under VcXsrv, so for now I've gone back to the windows version. Not sure if that's just how it is, or if I can make it run smoother, but I haven't had the time to look into it yet.
